compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -94,7 +94,8 @@ class Compiler {
          new LetOverApply,            // Lift blocks from receivers of applications
          new ArrayConstructors) ::   // Intercept creation of (non-generic) arrays and intrinsify.
     List(new Erasure) ::             // Rewrite types to JVM model, erasing all type parameters, abstract types and refinements.
-    List(new ElimErasedValueType,    // Expand erased value types to their underlying implmementation types
+    List(new PureStats,              // Remove pure stats from blocks
+         new ElimErasedValueType,    // Expand erased value types to their underlying implmementation types
```

Comment:
I would keep ElimErasedValueType the first phase after erasure, this way values with ErasedValueType as a type only exist for the minimal amount of time they need to.

Reply:
Done